### PR TITLE
Use pam_unix no_pass_expiry

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: man/pam_systemd.xml [hunk 1]
```diff
@@ -314,7 +314,7 @@ auth      required   pam_deny.so
 
 account   required   pam_nologin.so
 -account  sufficient pam_systemd_home.so
-account   sufficient pam_unix.so
+account   sufficient pam_unix.so no_pass_expiry
```
Comment (poettering) [inline man/pam_systemd.xml]: hmm, but the "sufficient" in the pam_systemd_home line means that this line isn't even tested anymore, hence this should not be needed

File: man/pam_systemd_home.xml [hunk 2]
```diff
@@ -135,7 +135,7 @@ auth      required   pam_deny.so
 
 account   required   pam_nologin.so
 <command>-account  sufficient pam_systemd_home.so</command>
-account   sufficient pam_unix.so
+account   sufficient pam_unix.so no_pass_expiry
```
Comment (poettering) [inline man/pam_systemd_home.xml]: ditto

File: man/pam_systemd_home.xml [hunk 3]
```diff
@@ -146,7 +146,7 @@ password  required   pam_deny.so
 -session  optional   pam_loginuid.so
 <command>-session  optional   pam_systemd_home.so</command>
 -session  optional   pam_systemd.so
-session   required   pam_unix.so</programlisting>
+session   required   pam_unix.so no_pass_expiry</programlisting>
```
Comment (poettering) [inline man/pam_systemd_home.xml]: looking at the sources, this sounds unnecessary, as the flag is not checked in the "session" hook, but only in the account hook

File: src/login/systemd-user.in [hunk 4]
```diff
@@ -5,7 +5,7 @@
 {% if ENABLE_HOMED %}
 -account sufficient pam_systemd_home.so
 {% endif %}
-account sufficient pam_unix.so
+account sufficient pam_unix.so no_pass_expiry
```
Comment (poettering) [inline src/login/systemd-user.in:8]: this definitely looks right, given we don't really care about passwords at all for this service, its a background service without any password prompting afte rall

**Comment by poettering** [comment]
first commit looks good to me, second one I am not so sure, see above

**Comment by keszybz** [comment]
OK, I'll drop the second commit. I tested the first one, so I'm pretty confident that it's doing the right thing. The second one was speculative.

**Pull request merged**